add multiconfig for launch workers

BillyWooo opened the pull request:
as the title.
Launch workers with [local parachain / staging parachain / production parachain]

resolves #1000

Kailai-Wang committed:
adjust docker tag for tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
use integritee/pallets master branch

Commit:
Merge remote-tracking branch 'origin/dev' into tee-dev

Commit:
Merge remote-tracking branch 'origin/dev' into tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
add teeracle to tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Commit:
Merge remote-tracking branch 'origin/dev' into tee-dev

Commit:
Merge remote-tracking branch 'origin/dev' into tee-dev

Kailai-Wang committed:
switch to sdk-v0.9.0-polkadot-v0.9.26 as release branch

Commit:
Merge branch 'tee-dev' of github.com:litentry/litentry-parachain into tee-dev

Commit:
Merge remote-tracking branch 'origin/dev' into tee-dev

Commit:
add back the dummy method for testing purpose

Commit:
fix make clippy

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
small cleanup

Kailai-Wang committed:
add back teeracle

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
use specific branch of tee-primitives (todo:revert it)

Kailai-Wang committed:
correctify branch for tee-primitives

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Kailai-Wang committed:
temporarily use TEE pallets master branch

Kailai-Wang committed:
add EnsureOrigin impl and tests

Kailai-Wang committed:
its better to place it in runtime_common

Kailai-Wang committed:
Merge branch '895-use-real-enclave-signer' into tee-dev

Kailai-Wang committed:
Merge branch 'dev' into tee-dev

Commit:
remove the dummy extrinsic (used for testing previous)

Commit:
rename ruleset -> assertion

Commit:
update renaming

Commit:
revert balance type

Kailai-Wang committed:
Cherry-Pick: 47386434
Use rococo as dev network (#918)
* add rococo staging
* cargo update
* update README
* fix bad english

Commit:
according to upstream, update pallets branch to 'develop'

Commit:
fix build error

Commit:
add rococo docker file

BillyWooo committed:
Update tee-dev branch into polkadot v0.9.29 (#971)

* merge tee-dev branch into dev branch (#902)

* adjust docker tag for tee-dev

* use integritee/pallets master branch

* add teeracle to tee-dev

* switch to sdk-v0.9.0-polkadot-v0.9.26 as release branch

* add back the dummy method for testing purpose

* fix make clippy

* small cleanup

* add back teeracle

* use specific branch of tee-primitives (todo:revert it)

* correctify branch for tee-primitives

* temporarily use TEE pallets master branch

* add EnsureOrigin impl and tests

* its better to place it in runtime_common

* remove the dummy extrinsic (used for testing previous)

Co-authored-by: Kailai Wang <Kailai.Wang@hotmail.com>
Co-authored-by: BillyWooo <yang@litentry.com>

* 0.9.29 update (#891)

* feat: rough update to 0.9.29

* debug:temp fmt

* debug: 0.9.29 without trystate

* debug: Integration with latest try-runtime

* debug: update weight template

* debug: fmt

* debug: patch toml try-runtime

* debug: fmt

* debug: revert template non-weight part

* debug: add teeracle try-runtime

* debug: cargo lock

* fix Cargo.lock (#904)

* lift the bridge transfer restriciton and bump version (#907)

* use hash to compute the raw payload (#910)

* remove Litmus sudo (#876)

* sudo

* change Root into Root +democracy

* debug: recover chainspec rootkey

* fix clippy

* try to fix #900

* add debug log #900

* make CI tests pass after remove litmus `Sudo`

* fmt

* debug

* debug

* rococo/litentry align with litmus

* remove useless test code

Co-authored-by: Kai <7630809+Kailai-Wang@users.noreply.github.com>
Co-authored-by: BillyWooo <yang@litentry.com>
Co-authored-by: zhiming-zhong <zhiming.zhong@litentry.com>

* Use rococo as dev network (#918)

* add rococo staging

* cargo update

* update README

* fix bad english

* Bump aes-gcm from `be0d6c2` to `5eb6d1a` (#925)

Bumps [aes-gcm](https://github.com/RustCrypto/AEADs) from `be0d6c2` to `5eb6d1a`.
- [Release notes](https://github.com/RustCrypto/AEADs/releases)
- [Commits](https://github.com/RustCrypto/AEADs/compare/be0d6c2355aa4dee288e18446e49d3765eaed207...5eb6d1a046371411d062db62671da29799beba7a)

---
updated-dependencies:
- dependency-name: aes-gcm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump scale-info from 2.2.0 to 2.3.0 (#926)

Bumps [scale-info](https://github.com/paritytech/scale-info) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/paritytech/scale-info/releases)
- [Changelog](https://github.com/paritytech/scale-info/blob/master/CHANGELOG.md)
- [Commits](https://github.com/paritytech/scale-info/commits/v2.3.0)

---
updated-dependencies:
- dependency-name: scale-info
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Minor update (#928)

* add dwellir bootnode

* bump litmus runtime version

* cargo update

* feat: relocate remove_sudo migration code (#936)

* feat: relocate remove_sudo migration code

* debug: fmt

* debug: change staking parameter (#931)

* debug: change staking parameter

* debug: bump rococo.litentry spec version

* Use sccache in Dockerfile (#937)

* try to switch to github-runner

* use sccache in Dockerfile for self-hosted runner

* fix syntax

* download sccache

* fix path and use pre-built taplo

* intentionally change lib.rs

* intentionally update a pallet only

* try to add/stop sccache server

* minor update

* update README

* try to use rust-cache

* use /root

* use GITHUB_REF_NAME in hash-key

* I dislike github syntax

* another try

* deprecate set-output

* revert back lib.rs

* dintinguish between release and prod

* fix RUN order

* use head_ref or ref_name

Co-authored-by: BillyWooo <yang@litentry.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: Kailai Wang <Kailai.Wang@hotmail.com>
Co-authored-by: BillyWooo <yang@litentry.com>
Co-authored-by: WMQ <46511820+wangminqi@users.noreply.github.com>
Co-authored-by: Kai <7630809+Kailai-Wang@users.noreply.github.com>
Co-authored-by: zhiming-zhong <zhiming.zhong@litentry.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
add BUILDKIT explicitly

Kailai-Wang committed:
Merge tee worker repo (#987)

* initial adding tee-worker src

* fix link path

* adjust path and content

* fix path

* fmt

* add ignore/exclude path

* fix the path for upload/download artifacts

* fix path

Kailai-Wang committed:
Switch to local dependencies (#1037)

* use local dependencies

* fix docker path

Kailai-Wang committed:
Try to fix the docker image pushing (#1040)

* debug: try download and push

* add debug commands

* remove file checker

* remove defaults

* add pwd

* make sure to chekcout code

* checkout code

* add CI yml to detect list

Kailai-Wang committed:
remove unneeded CI steps (#1049)

0xverin committed:
tee-worker ts-tests (#1039)

* add cross-env&&dotenv

* set env

* random shard for testing

* Change the versions of @polkadot

* change the version of @polkadot/types

* rm yarn.lock

BillyWooo committed:
add multiconfig for launch workers with local development / staging env / production env

jingleizhang commented:
1. https://github.com/litentry/litentry-parachain/blob/504196026a0719baa9e75a64775a24199e41c41f/tee-worker/litentry/core/data-providers/src/lib.rs#L59-L65

2. https://github.com/litentry/litentry-parachain/blob/504196026a0719baa9e75a64775a24199e41c41f/tee-worker/litentry/core/data-providers/src/twitter_official.rs#L103

3. https://github.com/litentry/litentry-parachain/blob/504196026a0719baa9e75a64775a24199e41c41f/tee-worker/litentry/core/data-providers/src/discord_official.rs#L71

Those hardcodes should be moved to config files.

0xverin committed:
Ts tests (#1053)

* add cross-env&&dotenv

* set env

* random shard for testing

* Change the versions of @polkadot

* change the version of @polkadot/types

* rm yarn.lock

* change yarn.lock

* generateChallengeCode

* modify yarn run command

* change defaultSinger

* add getSinger

* add Sign functions

* add getSinger && generateChallengeCode

* add ID_HUB_URL for sign message

* modify defaultSigner

* modify shard

BillyWooo committed:
add comment

Kailai-Wang committed:
Add more testcases for IMP mock (#1057)

* add more testcases

* fix compile

* fix file checker in tee-worker-ci

Co-authored-by: ericzhang <jingleizhang@users.noreply.github.com>

BillyWooo committed:
982 renaming issue (#1052)

* rename `link_identity` to `create_identity`

* rename `unlink_identity` to `remove_identity`

* add up the missing ones

* add up missing ones 2

* make fmt happy

* hope not more missed...

* remove not needed file

* update old name from merge (link -> create)

Commit:
Merge branch 'tee-dev' into 1000_muti_config_to_launch_workers

BillyWooo committed:
Merge branch '1000_muti_config_to_launch_workers' of github.com:litentry/litentry-parachain into 1000_muti_config_to_launch_workers

BillyWooo committed:
move hardcode const to input parameter

BillyWooo committed:
make clippy happy

Kailai-Wang commented:
As mentioned it might be better to directly extend the `tee-worker/local-setup/github-action-config.json` and add more such JSONs for different envs. Keep it simple.

But please discuss it directly with @jingleizhang

jingleizhang commented:
@Kailai-Wang  How can we keep those prod config files safe, through our CI tools or platforms(e.g, ansible/Rundeck)?

Kailai-Wang commented:
> @Kailai-Wang How can we keep those prod config files safe, through our CI tools or platforms(e.g, ansible/Rundeck)?

similar to our collators.
Even in the prod config I don't think there are too many secrets (except API keys maybe?), servers won't be reachable from outside. We can manually copy the files/edit the secrets there, it should be an one-time effort as it's unlikely to change. Alternatively we can use something like git crypt.

Maybe let's discuss it offline instead of in a public thread :)

0xverin committed:
link web3 and verify web3 (#1072)

* add cross-env&&dotenv

* set env

* random shard for testing

* Change the versions of @polkadot

* change the version of @polkadot/types

* rm yarn.lock

* change yarn.lock

* generateChallengeCode

* modify yarn run command

* change defaultSinger

* add getSinger

* add Sign functions

* add getSinger && generateChallengeCode

* add ID_HUB_URL for sign message

* modify defaultSigner

* modify shard

* add ethers provider&&wallet

* add eth endpoint

* add @ethersproject/providers&&ethers

* substrate&&ethereum tests

* web3 types

* nonce

* change methods name

* change methods name

* change events

* change methods name

* change event

* modify method name

jingleizhang commented:
Get it.

Kailai-Wang committed:
Emit new challenge code for each `create_identity` (#1074)

* some more renaming

* adjustment to IMP mock

* change to sgx-runtime pallets

jingleizhang commented on tee-worker/litentry/core/data-providers/Cargo.toml:
```diff
@@ -14,6 +14,7 @@ thiserror = { version = "1.0.26", optional = true }
 url = { version = "2.0.0", optional = true }
 
 # no_std dependencies
+lazy_static = { version = "1.1.0", features = ["spin_no_std"] }
```
the latest version is 1.4.0

jingleizhang commented on tee-worker/litentry/core/data-providers/src/lib.rs:
```diff
@@ -55,22 +61,67 @@ pub mod twitter_official;
-	pub(crate) const DISCORD_LITENTRY: &str = "http://localhost:9527";
+lazy_static! {
+	pub static ref G_DATA_PROVIDERS: RwLock<DataProvidersStatic> =
+		RwLock::new(DataProvidersStatic::new());
```
Why the RwLock is needed here?

BillyWooo replied:
to avoid race condition and `unsafe` code block.

jingleizhang commented on tee-worker/scripts/launch_local_worker.sh:
```diff
@@ -1,80 +1,126 @@
+--untrusted-worker-port ${untrusted_worker_port} \
+--twitter-official-url ${TWITTER_OFFICIAL_URL} \
+--twitter-litentry-url ${TWITTER_LITENTRY_URL} \
+--twitter-auth-token ${TWITTER_AUTH_TOKEN} \
```
Those values should be directly loaded from config files, and should not be passed in as command-line arguments. For example, if we want to add a new config field, we don't need to change the launch script again.

jingleizhang committed:
Merge branch 'tee-dev' into 1000_muti_config_to_launch_workers

Kailai-Wang committed:
Return encrypted IDGraph when changed (#1080)

* add id_graph to ocall extrinsics

* adjust callback extrinsic

* add cross-env&&dotenv

* set env

* random shard for testing

* Change the versions of @polkadot

* change the version of @polkadot/types

* rm yarn.lock

* slight renaming

* change maxVerificationDelay to 30min

* change yarn.lock

* generateChallengeCode

* modify yarn run command

* change defaultSinger

* add getSinger

* add Sign functions

* add getSinger && generateChallengeCode

* add ID_HUB_URL for sign message

* modify defaultSigner

* modify shard

* add ethers provider&&wallet

* add eth endpoint

* add @ethersproject/providers&&ethers

* substrate&&ethereum tests

* web3 types

* nonce

* change methods name

* change methods name

* change events

* change methods name

* change event

* modify method name

* ts-test support `id_graph`

* fix bug

* remove log

* resolve conflicts

Co-authored-by: Verin1005 <daqingchong0809@gmail.com>
Co-authored-by: zzz <zhiming.zhong@litentry.com>

BillyWooo committed:
Merge remote-tracking branch 'origin/tee-dev' into 1000_muti_config_to_launch_workers

BillyWooo committed:
Tee worker polkadot v0.9.32 (#1093)

* merge tee-dev branch into dev branch (#902)

* adjust docker tag for tee-dev

* use integritee/pallets master branch

* add teeracle to tee-dev

* switch to sdk-v0.9.0-polkadot-v0.9.26 as release branch

* add back the dummy method for testing purpose

* fix make clippy

* small cleanup

* add back teeracle

* use specific branch of tee-primitives (todo:revert it)

* correctify branch for tee-primitives

* temporarily use TEE pallets master branch

* add EnsureOrigin impl and tests

* its better to place it in runtime_common

* remove the dummy extrinsic (used for testing previous)

Co-authored-by: Kailai Wang <Kailai.Wang@hotmail.com>
Co-authored-by: BillyWooo <yang@litentry.com>

* 0.9.29 update (#891)

* feat: rough update to 0.9.29

* debug:temp fmt

* debug: 0.9.29 without trystate

* debug: Integration with latest try-runtime

* debug: update weight template

* debug: fmt

* debug: patch toml try-runtime

* debug: fmt

* debug: revert template non-weight part

* debug: add teeracle try-runtime

* debug: cargo lock

* fix Cargo.lock (#904)

* lift the bridge transfer restriciton and bump version (#907)

* use hash to compute the raw payload (#910)

* remove Litmus sudo (#876)

* sudo

* change Root into Root +democracy

* debug: recover chainspec rootkey

* fix clippy

* try to fix #900

* add debug log #900

* make CI tests pass after remove litmus `Sudo`

* fmt

* debug

* debug

* rococo/litentry align with litmus

* remove useless test code

Co-authored-by: Kai <7630809+Kailai-Wang@users.noreply.github.com>
Co-authored-by: BillyWooo <yang@litentry.com>
Co-authored-by: zhiming-zhong <zhiming.zhong@litentry.com>

* Use rococo as dev network (#918)

* add rococo staging

* cargo update

* update README

* fix bad english

* Bump aes-gcm from `be0d6c2` to `5eb6d1a` (#925)

Bumps [aes-gcm](https://github.com/RustCrypto/AEADs) from `be0d6c2` to `5eb6d1a`.
- [Release notes](https://github.com/RustCrypto/AEADs/releases)
- [Commits](https://github.com/RustCrypto/AEADs/compare/be0d6c2355aa4dee288e18446e49d3765eaed207...5eb6d1a046371411d062db62671da29799beba7a)

---
updated-dependencies:
- dependency-name: aes-gcm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump scale-info from 2.2.0 to 2.3.0 (#926)

Bumps [scale-info](https://github.com/paritytech/scale-info) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/paritytech/scale-info/releases)
- [Changelog](https://github.com/paritytech/scale-info/blob/master/CHANGELOG.md)
- [Commits](https://github.com/paritytech/scale-info/commits/v2.3.0)

---
updated-dependencies:
- dependency-name: scale-info
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Minor update (#928)

* add dwellir bootnode

* bump litmus runtime version

* cargo update

* feat: relocate remove_sudo migration code (#936)

* feat: relocate remove_sudo migration code

* debug: fmt

* debug: change staking parameter (#931)

* debug: change staking parameter

* debug: bump rococo.litentry spec version

* Use sccache in Dockerfile (#937)

* try to switch to github-runner

* use sccache in Dockerfile for self-hosted runner

* fix syntax

* download sccache

* fix path and use pre-built taplo

* intentionally change lib.rs

* intentionally update a pallet only

* try to add/stop sccache server

* minor update

* update README

* try to use rust-cache

* use /root

* use GITHUB_REF_NAME in hash-key

* I dislike github syntax

* another try

* deprecate set-output

* revert back lib.rs

* dintinguish between release and prod

* fix RUN order

* use head_ref or ref_name

Co-authored-by: BillyWooo <yang@litentry.com>

* Update v0.9.30 (#942)

* some deps update to v0.9.30

* deps update to v0.9.30

* format code

* change deps

* fix docker file

* fix docker file

* change docker file

* change docker file

* fix docker yml file

* fix docker yml file

* fix docker yml file

* add install dependencis shell

* fix install dependencis shell

* fix install dependencis shell

* fix docker file

* fix yml file

* add sudo permisson

* add some dep

Co-authored-by: Chengcheng Shi <Chengcheng@litentry>

* optimize cli commands (#947)

* optimize cli commands

* Change some command's names

* format docker file

Co-authored-by: SCC <root@LAPTOP-73QQ3GCK.localdomain>

* Add onfinality script (#952)

* minor update

* bump version and cargo update

* [benchmarking bot] Auto commit generated weights files (#951)

Co-authored-by: Kailai-Wang <Kailai-Wang@users.noreply.github.com>

* try to use new weights

* fix typo

* add missing mods

* fix compile

* add onFinality image version update

* fix indention

* add continue-on-error

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Kailai-Wang <Kailai-Wang@users.noreply.github.com>

* Enable parachain staking extrinsic (#955)

* debug: change extrinsic filter

* debug: fmt and clippy

* debug: revert cargo.lock

* Add release bot GHA (#957)

* add release bot yml

* small update

* Issue 958 (#959)

* fix parachain staking benchmark code

* fix syntax

* omit CollatorSelection benchmark code

* [benchmarking bot] Auto commit generated weights files (#961)

Co-authored-by: Jinsipang <Jinsipang@users.noreply.github.com>

Co-authored-by: Chengcheng Shi
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Bump serde_json from 1.0.87 to 1.0.88 (#967)

Bumps [serde_json](https://github.com/serde-rs/json) from 1.0.87 to 1.0.88.
- [Release notes](https://github.com/serde-rs/json/releases)
- [Commits](https://github.com/serde-rs/json/compare/v1.0.87...v1.0.88)

---
updated-dependencies:
- dependency-name: serde_json
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Optimize parachain-staking benchmark code to shorten running time (#974)

* remove reward for delegator warn logs

* fix benchmark shell

* fix some syntax

* fix shell

* fix shells

* optimized benchmark code and shell

* fix some params

* optimize benchmark code

* [benchmarking bot] Auto commit generated weights files (#973)

Co-authored-by: Jinsipang

* fix benchmark code

* add staking benchmark code at runtime

* fix benchmark code

* fix benchmark code

* fix benchmark code

* format code

* change benchmark code args

Co-authored-by: Chengcheng Shi <chengcheng@litentry.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Bump aes-gcm from `86250ac` to `6ac2d1c` (#970)

Bumps [aes-gcm](https://github.com/RustCrypto/AEADs) from `86250ac` to `6ac2d1c`.
- [Release notes](https://github.com/RustCrypto/AEADs/releases)
- [Commits](https://github.com/RustCrypto/AEADs/compare/86250ac216bb0230c178765303265c6f037d9408...6ac2d1cec6f6658ed5fbb1befa93a370d567f354)

---
updated-dependencies:
- dependency-name: aes-gcm
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump serde from 1.0.147 to 1.0.148 (#991)

Bumps [serde](https://github.com/serde-rs/serde) from 1.0.147 to 1.0.148.
- [Release notes](https://github.com/serde-rs/serde/releases)
- [Commits](https://github.com/serde-rs/serde/compare/v1.0.147...v1.0.148)

---
updated-dependencies:
- dependency-name: serde
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump serde_json from 1.0.88 to 1.0.89 (#990)

Bumps [serde_json](https://github.com/serde-rs/json) from 1.0.88 to 1.0.89.
- [Release notes](https://github.com/serde-rs/json/releases)
- [Commits](https://github.com/serde-rs/json/compare/v1.0.88...v1.0.89)

---
updated-dependencies:
- dependency-name: serde_json
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* feat: add PolkadotXCM filter (#993)

* update Polkadot v0.9.32 (#1043)

* update deps to polkaodt-v0.9.31

* update node cli

* fix some test code

* update to polkadot-v0.9.32

* fix buuld benchmark failed and run test failed issues

* fix node code

* [benchmarking bot] Auto commit generated weights files (#1046)

Co-authored-by: Chengcheng Shi <chengcheng@litentry.com>

* Support standalone node for parachain (#1059)

* use unified crate version and cargo update

* initial add standalone mode

* fix chain spec id

* fix clippy

* add script to start standalone node

* add Makefile

* use 9944 as parachain port consistently

* adjust bridge endpoint

* minor update

* recover docker image

* minor update

* bump version

* update README

* Bump decode-uri-component from 0.2.0 to 0.2.2 in /ts-tests (#1060)

Bumps [decode-uri-component](https://github.com/SamVerschueren/decode-uri-component) from 0.2.0 to 0.2.2.
- [Release notes](https://github.com/SamVerschueren/decode-uri-component/releases)
- [Commits](https://github.com/SamVerschueren/decode-uri-component/compare/v0.2.0...v0.2.2)

---
updated-dependencies:
- dependency-name: decode-uri-component
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Backport from release-v0.9.13 banch (#1067)

* minor fix of text search

* fix small typo

* Bump express from 4.17.2 to 4.18.2 in /ts-tests (#1071)

Bumps [express](https://github.com/expressjs/express) from 4.17.2 to 4.18.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.17.2...4.18.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Runtime upgrade simulation (#1081)

* update script

* use relative path

* update script

* improve arg handling

* fix trap

* simulmation runtime upgrade demo impl

* runtime-upgrade ts impl

* fix some syntax

* fix makefile

* optimized scripts && cli

* use master branch of fork-off-substrate

* optimize ts code

* fix description

* fix action sytanx

* fix cli syntax

* fix cli syntax

* fix cli syntax

* fix cli syntax

* fix cli syntax

* fix cli syntax

* fix cli syntax

* fix cli syntax

* optimize ts-code & cli

* adjustment of script and GHA

* try to fix CI

Co-authored-by: Scc <root@LAPTOP-73QQ3GCK.localdomain>
Co-authored-by: Scc <chengcheng@litentry.com>

* update dependencies from fork pallets to origin integritee pallets v0.9.32 (#1083)

* switch to temp branch

* update to integritee-pallets v0.9.32

* update pallets repo source

* add up missing lock file

* [benchmarking bot] Auto commit generated weights files

* [benchmarking bot] Auto commit generated weights files (#1082)

Co-authored-by: BillyWooo <BillyWooo@users.noreply.github.com>

* Revert "[benchmarking bot] Auto commit generated weights files (#1082)"

This reverts commit 6b11ff212ce5f047f2172eda10be9748d888570e.

* Revert "Merge remote-tracking branch 'origin/benchmarking-bot-3656108355' into pr-integritee-pallets-v0.9.32"

This reverts commit 3cb2961b5246083a257de0f8d4a17f5325e4fb07, reversing
changes made to d93c2447c12db33dcd86243ce54d58d3484eea6e.

Co-authored-by: BillyWooo <BillyWooo@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: zhizming-zhong <zhiming.zhong@litentry.com>

* Bump serde from 1.0.149 to 1.0.150 (#1085)

Bumps [serde](https://github.com/serde-rs/serde) from 1.0.149 to 1.0.150.
- [Release notes](https://github.com/serde-rs/serde/releases)
- [Commits](https://github.com/serde-rs/serde/compare/v1.0.149...v1.0.150)

---
updated-dependencies:
- dependency-name: serde
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Bump scale-info from 2.3.0 to 2.3.1 (#1086)

Bumps [scale-info](https://github.com/paritytech/scale-info) from 2.3.0 to 2.3.1.
- [Release notes](https://github.com/paritytech/scale-info/releases)
- [Changelog](https://github.com/paritytech/scale-info/blob/master/CHANGELOG.md)
- [Commits](https://github.com/paritytech/scale-info/compare/v2.3.0...v2.3.1)

---
updated-dependencies:
- dependency-name: scale-info
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* Add test local pallets benchmarking shell (#1087)

* add local benchmark testing shell

* fix benmarking scripts

* fix some syntax

* fix some syntax

* fix some syntax

Co-authored-by: SCC <chengcheng@litentry.com>

* Revert "Add test local pallets benchmarking shell (#1087)" (#1091)

This reverts commit 1c1ee636674925b99cbdd621924cc09bf3445744.

* update tee-worker to upstream/master (3cc02342)

* bump up version to 0.9.32

* cargo build / fmt / clippy / test pass

* parachain build

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: Kailai Wang <Kailai.Wang@hotmail.com>
Co-authored-by: BillyWooo <yang@litentry.com>
Co-authored-by: WMQ <46511820+wangminqi@users.noreply.github.com>
Co-authored-by: Kai <7630809+Kailai-Wang@users.noreply.github.com>
Co-authored-by: zhiming-zhong <zhiming.zhong@litentry.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: SCC <87012234+Jinsipang@users.noreply.github.com>
Co-authored-by: Chengcheng Shi <Chengcheng@litentry>
Co-authored-by: SCC <root@LAPTOP-73QQ3GCK.localdomain>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Kailai-Wang <Kailai-Wang@users.noreply.github.com>
Co-authored-by: Chengcheng Shi <chengcheng@litentry.com>
Co-authored-by: BillyWooo <BillyWooo@users.noreply.github.com>
Co-authored-by: ericzhang <jingleizhang@users.noreply.github.com>

BillyWooo committed:
Merge remote-tracking branch 'origin/dev' into tee-dev

BillyWooo committed:
update tee worker ci

BillyWooo committed:
Merge remote-tracking branch 'origin/tee-dev' into 1000_muti_config_to_launch_workers

BillyWooo committed:
Merge remote-tracking branch 'origin/dev' into 1000_muti_config_to_launch_workers

BillyWooo committed:
refactor terminal parameter into json config file

BillyWooo committed:
update dependency version

BillyWooo committed:
rm not used files

BillyWooo committed:
fix path

BillyWooo committed:
fix config file path issue inside CI

Kailai-Wang commented on tee-worker/service/src/cli.yml:
```diff
@@ -80,9 +80,14 @@ args:
+        help: Cleans and purges any previous state and key files and generates them anew before starting.
+    - running-mode:
+        long: running-mode
+        help: Litentry TEE service running mode <dev/stage/prod>
```
I think staging is the conventional word 😂, same for `worker_config_*`

BillyWooo replied:
🤣

Kailai-Wang commented on tee-worker/scripts/launch.sh:
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
```
when shall we run `launch.sh` and when `launch.py`?

BillyWooo replied:
Both are fine. `launch.sh` would be more preferred for local launch & test. `launch.py` + `config.json` would have more flexibility on ports parameter inputs (`launch.sh` will just increase the port number between different worker. `config.json` can give any port number).

Kailai-Wang commented on tee-worker/local-setup/github-action-config-rococo.json:
```diff
@@ -15,7 +15,9 @@
-        "443"
+        "443",
+        "running-mode",
+        "stage"
```
I think `-u` and `-m` can be confusing/conflicting: can I use `-u wss://parachain-eu-0.staging.litentry.io` and `-m dev` at the same time? Again it depends what are `dev/staging/prod` supposed to do

BillyWooo replied:
I think there is no shortcut for 'running-mode'. Always need to type the full name. I also feel confusing about these parameter shortcuts. That's the reason NO shortcut provided here.

Kailai-Wang commented on tee-worker/local-setup/worker_config_dev.json:
```diff
@@ -0,0 +1,8 @@
+{
```
I don't see a difference between dev/staging/prod, this needs a clear definition: what exactly should differ in these different configs? -- needs to be evaluated/defined by who opens this issue (maybe @jingleizhang)

BillyWooo replied:
Yes. For the moment. These three files are almost dummy placeholders. Need to define/clarify later.

Kailai-Wang reviewed (commented):
Overall it looks better, thanks.

The biggest concern is still what should be defined in `dev/staging/prod` configs -- I'll forward this to @jingleizhang

BillyWooo committed:
Merge remote-tracking branch 'origin/dev' into 1000_muti_config_to_launch_workers

BillyWooo committed:
renaming stage -> staging

jingleizhang reviewed (approved):
At least we have those prod/staging/dev config files so that we can work with the same application code in different environments. 

All parameters should be moved into the config file, and only one parameter that represents the env needs to be kept, actually. But for now, we're just using the config to save some no-more hard-coded information. At least it helps us keep some api keys safe.

Please correct me if I'm wrong: `prod` stands for `Litentry`, `staging` means `parachain-eu-0.staging.litentry.io`, and `dev` means `rococo`. 

I think we could change those names to something more clear if we think it's necessary. 😂
